bowtie/_report.py

```diff
@@ -155,6 +156,26 @@ def failed_fast(self, seq_case: SeqCase):
             "Stopping -- the maximum number of unsuccessful tests was reached",
         )
 
+    def failed_validate_schema_and_dialect(self, schema: Any, dialect: Any):
+        # here we will raise the exception for the schema and dialect conflict
+
+        if "$schema" in schema:
+            schema = schema["$schema"]
+            schema_dialect = Dialect.by_uri().get(URL.parse(schema))
```

Comment:
There's also no reason to do this parsing, you simply want to do `if dialect.uri != schema_dialect` in all these places.